src/component/addons/AddonList/AvailableAddons/AvailableAddons.tsx

```diff
@@ -10,13 +10,30 @@ import {
 } from '@material-ui/core';
 import ConditionallyRender from '../../../common/ConditionallyRender/ConditionallyRender';
 import { CREATE_ADDON } from '../../../providers/AccessProvider/permissions';
-import PropTypes from 'prop-types';
+import { useHistory } from 'react-router-dom';
+import AccessContext from '../../../../contexts/AccessContext';
 
-const AvailableAddons = ({ providers, getIcon, hasAccess, history }) => {
-    
-    const renderProvider = provider => (
+interface IProvider {
+    name: string;
+    displayName: string;
+    description: string;
+    documentationUrl: string;
+    parameters: object[];
+    events: string[];
+}
+
+interface IAvailableAddonsProps {
+    AddonIcon: (name: string) => ReactElement;
+    providers: IProvider[];
+}
+
+const AvailableAddons = ({ providers, AddonIcon }: IAvailableAddonsProps) => {
+    const history = useHistory();
+    const { hasAccess } = useContext(AccessContext);
+
+    const renderProvider = (provider: IProvider) => (
         <ListItem key={provider.name}>
-            <ListItemAvatar>{getIcon(provider.name)}</ListItemAvatar>
+            <ListItemAvatar>{AddonIcon(provider.name)}</ListItemAvatar>
```

Comment:
`AddonIcon` suggests to me that this is a component, when in fact it's a function. Could we rename it `getAddonIcon` ?